src/js/containers/FormPage.jsx

```diff
@@ -63,14 +63,16 @@ class FormPage extends React.Component {
 
     const path = getNextPagePath(route.pageList, form.data, location.pathname);
 
-    this.props.router.push(path);
+    // this.props.router.push(path);
+    this.props.route.goToRoute(path);
   }
 
   goBack = () => {
     const { form, route: { pageList }, location } = this.props;
     const path = getPreviousPagePath(pageList, form.data, location.pathname);
 
-    this.props.router.push(path);
+    // this.props.router.push(path);
+    this.props.route.goToRoute(path);
```

Comment:
I need a way to pass a function to `<FormPage/>` from the starter app that will push up the next path we want to go to to the starter app. Once the starter app has the next path, it can determine the routing to get there. The tricky part is how to create this hook from the starter app side. `<FormPage/>` is rendered as the component for a particular route, not directly rendered in a parent component, so it gets a little tricky to pass props to it.

Reply:
I think the way to do this is to have the app have a component, say `WholePage`, that gets `withRouter`ed. That component puts together the parts of the page that the app wants, including the `FormPage` but also a header, footer, sidebar, or whatever. Any of that can be route specific. When `WholePage` renders it passes the current route (or page key or whatever lets USFS find the right page) to `FormPage`.  Note that this gives the app flexibility over layout if they want it, although I'd expect our `WholePage` implementation does what we do today except that we'd put the `formConfig.footer` in it.

Reply:
Yes that's what I was thinking, but also including the navigation buttons. This moves us towards a USFS library that is focused on rendering and validating form fields, with basically nothing else. All of the surrounding structure, like header, footer, navigation, etc., exists in the starter app. Just confirming that's the direction we think it makes sense to go in?